static/app/utils/discover/discoverOrMetricsQuery.tsx

```diff
@@ -0,0 +1,43 @@
+import {
+  convertDiscoverFieldsToMetrics,
+  convertDiscoverFieldsToMetricsGroupBys,
+} from 'sentry/utils/metrics/fields';
+import MetricsRequest from 'sentry/utils/metrics/metricsRequest';
+import {useMetricsSwitch} from 'sentry/views/performance/metricsSwitch';
+
+import withApi from '../withApi';
+
+import DiscoverQuery, {DiscoverQueryPropsWithThresholds} from './discoverQuery';
+
+function DiscoverOrMetricsQuery(props: DiscoverQueryPropsWithThresholds) {
+  const {isMetricsData} = useMetricsSwitch(); // TODO(metrics): decision logic will be dynamic in the future, not switch toggle
+
+  if (isMetricsData) {
+    const {eventView, orgSlug, api} = props;
+    const {start, end, statsPeriod, project, environment} = eventView;
+    const fields = eventView.fields.map(({field}) => field);
+
+    return (
+      <MetricsRequest
+        api={api}
+        orgSlug={orgSlug}
+        start={start}
+        end={end}
+        statsPeriod={statsPeriod}
+        project={project}
+        environment={environment}
+        // query={eventView.getQueryWithAdditionalConditions()} TODO(metrics)
```

Comment:
Any reason we do not plug in

```javascript
query={new MutableSearch(eventView.query).formatString()}
```

here, as for other instances of `MetricRequest`?

Reply:
Not really, I didn't expect we would need it for testing the POC. I added it there 👍